torch/testing/_internal/logging_tensor.py

```diff
@@ -114,14 +120,19 @@ def log_input(name: str, var: object):
     logging.getLogger("LoggingTensor").info("input", (name,), {}, (var,))
 
 @contextlib.contextmanager
-def capture_logs() -> Iterator[List[str]]:
+def capture_logs(is_dispatch_mode=False) -> Iterator[List[str]]:
     logger = logging.getLogger("LoggingTensor")
     log_list: List[str] = []
-    handler = LoggingTensorHandler(log_list)
+    handler = LoggingTensorHandler(log_list, use_shortid_for_all_tensors=is_dispatch_mode)
     logger.addHandler(handler)
     logger.setLevel(logging.INFO)
     logger.propagate = False
     try:
         yield log_list
     finally:
         logger.removeHandler(handler)
+
+@contextlib.contextmanager
+def capture_logs_with_logging_tensor_mode():
+    with push_torch_dispatch_mode(LoggingTensorMode(inner=None)), capture_logs(True) as logs:
```

Comment:
TODO: shouldn't don't need inner

Reply:
@samdow yeah this looks like we should just support direct constructor lol

Reply:
We should also add a test for this function

I'm pretty sure that this will error right now for the inner reason. Also flags to me that I really dislike that the argument for `push_torch_dispatch_mode` is going to be a constructor for a Mode while `enable_torch_dispatch_mode` is going to be the Mode itself. TODO (@samdow): fix that